pkg/registry/clusteroperator/cluster/strategy.go

```diff
@@ -91,6 +91,7 @@ func (clusterRESTStrategy) PrepareForCreate(ctx genericapirequest.Context, obj r
 	// Creating a brand new object, thus it must have no
 	// status. We can't fail here if they passed a status in, so
 	// we just wipe it clean.
+	cluster.Finalizers = []string{clusteroperator.FinalizerClusterOperator}
```

Comment:
This is causing problems for the TestCreate function in pkg/registry/clusteroperator/cluster/storage_test.go. Thinking about this more, I think it is bad design to add the finalizer in the registry strategy. That requires that there is a controller that will come along and take responsibility for the finalizer. I think it is better to have that controller add the finalizer itself.

I recommend we move adding the finalizer to [here](https://github.com/joelddiaz/cluster-operator/blob/16c22dc5813813272759b0cbeda577f219342825/pkg/controller/infra/infra_controller.go#L527). Add the finalizer if it is not already there.